nameserver/http.go

```diff
@@ -65,6 +67,17 @@ func ServeHTTP(listener net.Listener, version string, server *DNSServer, domain
 					return
 				} // oh, I already know this. whatever.
 			}
+
+			if dockerCli != nil {
+				if docker.IsContainerId(idStr) {
+					Debug.Printf("[http] '%s' does not seem a container id", idStr)
+				} else if running, err := dockerCli.IsContainerRunning(idStr); err != nil {
+					Error.Printf("[http] Could not check container status: %s", err)
+				} else if !running {
+					Info.Printf("[http] '%s' is not running: removing", idStr)
+					server.Zone.DeleteRecords(idStr, name, ip)
+				}
+			}
 		} else {
 			Info.Printf("[http] Ignoring name %s, not in %s", name, domain)
 		}
```

Comment:
The *intention* is to break the flow. And it only costs one line, and reduces indentation of 7.

It's not a big deal though. Leave it as is if you must.